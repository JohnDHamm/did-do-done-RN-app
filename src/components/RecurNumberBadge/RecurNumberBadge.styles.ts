import styled from 'styled-components/native';
import { COLORS, FONTS } from '../../styles';

export const Badge = styled.View<{ color: string }>`
  width: 32px;
  height: 32px;
  border-radius: 16px;
  background-color: ${(props) => props.color};
  justify-content: center;
  align-items: center;
`;

export const BadgeNumber = styled.Text`
  font-family: ${FONTS.PRIMARY};
  font-size: 20px;
  color: ${COLORS.WHITE};
`;
