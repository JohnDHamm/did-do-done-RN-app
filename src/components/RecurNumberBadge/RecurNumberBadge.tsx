import React from 'react';
import { COLORS } from '../../styles';
import { Badge, BadgeNumber } from './RecurNumberBadge.styles';

interface RecurNumberBadgeProps {
  total: number;
  type: 'missed' | 'today' | 'thisweek' | 'next30';
}
const RecurNumberBadge: React.FC<RecurNumberBadgeProps> = ({
  total,
  type,
}: RecurNumberBadgeProps) => {
  let color = '';

  switch (type) {
    case 'missed':
      color = COLORS.MISSED_RED;
      break;
    case 'today':
      color = COLORS.TODAY_GREEN;
      break;
    case 'thisweek':
      color = COLORS.THIS_WEEK_YELLOW;
      break;
    case 'next30':
      color = COLORS.PRIMARY_GRAY;
      break;
  }
  return (
    <Badge color={color}>
      <BadgeNumber>{total}</BadgeNumber>
    </Badge>
  );
};

export default RecurNumberBadge;
