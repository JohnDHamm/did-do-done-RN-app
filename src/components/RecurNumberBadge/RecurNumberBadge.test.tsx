import React from 'react';
import { cleanup } from 'react-native-testing-library';
import RecurNumberBadge from './RecurNumberBadge';
import { renderWithLightTheme } from '../../testing/helpers';

describe('RecurNumberBadge', () => {
  afterEach(cleanup);

  it('renders correctly with "missed" type', () => {
    const testRenderer = renderWithLightTheme(
      <RecurNumberBadge total={42} type="missed" />
    ).toJSON();

    expect(testRenderer).toMatchSnapshot();
  });

  it('renders correctly with "today" type', () => {
    const testRenderer = renderWithLightTheme(
      <RecurNumberBadge total={43} type="today" />
    ).toJSON();

    expect(testRenderer).toMatchSnapshot();
  });

  it('renders correctly with "thisweek" type', () => {
    const testRenderer = renderWithLightTheme(
      <RecurNumberBadge total={44} type="thisweek" />
    ).toJSON();

    expect(testRenderer).toMatchSnapshot();
  });

  it('renders correctly with "next30" type', () => {
    const testRenderer = renderWithLightTheme(
      <RecurNumberBadge total={45} type="next30" />
    ).toJSON();

    expect(testRenderer).toMatchSnapshot();
  });
});
