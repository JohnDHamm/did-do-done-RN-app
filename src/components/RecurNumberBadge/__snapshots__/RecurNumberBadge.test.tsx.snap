// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecurNumberBadge renders correctly with "missed" type 1`] = `
<View
  color="#B70000"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#B70000",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "Lobster-Regular",
          "fontSize": 20,
        },
      ]
    }
  >
    42
  </Text>
</View>
`;

exports[`RecurNumberBadge renders correctly with "next30" type 1`] = `
<View
  color="#c3c3c3"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#c3c3c3",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "Lobster-Regular",
          "fontSize": 20,
        },
      ]
    }
  >
    45
  </Text>
</View>
`;

exports[`RecurNumberBadge renders correctly with "thisweek" type 1`] = `
<View
  color="#ffd764"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffd764",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "Lobster-Regular",
          "fontSize": 20,
        },
      ]
    }
  >
    44
  </Text>
</View>
`;

exports[`RecurNumberBadge renders correctly with "today" type 1`] = `
<View
  color="#237b59"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#237b59",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "Lobster-Regular",
          "fontSize": 20,
        },
      ]
    }
  >
    43
  </Text>
</View>
`;
